crates/ra_ide/src/completion/complete_scope.rs

```diff
@@ -1,13 +1,20 @@
 //! Completion of names from the current scope, e.g. locals and imported items.
 
 use crate::completion::{CompletionContext, Completions};
+use hir::{ModuleDef, ScopeDef};
 
 pub(super) fn complete_scope(acc: &mut Completions, ctx: &CompletionContext) {
-    if !ctx.is_trivial_path {
+    if !ctx.is_trivial_path && !ctx.is_pat_binding_and_path {
         return;
     }
 
-    ctx.scope().process_all_names(&mut |name, res| acc.add_resolution(ctx, name.to_string(), &res));
+    ctx.scope().process_all_names(&mut |name, res| match (ctx.is_pat_binding_and_path, &res) {
+        (true, ScopeDef::ModuleDef(ModuleDef::Function(..))) => (),
+        (true, ScopeDef::ModuleDef(ModuleDef::Static(..))) => (),
+        (true, ScopeDef::ModuleDef(ModuleDef::Const(..))) => (),
```

Comment:
Constants should be allowed. In `match foo { None => (), ...}`, `None` is more or less a constant